### PR TITLE
Dịch "softmax-regression-scratch_vn" - Phần 6

### LINKED ISSUE #772: Dịch "softmax-regression-scratch_vn" - Phần 6
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/softmax-regression-scratch_vn.md#L455-L515

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by azraelzhor**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #772

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 1]
```diff
@@ -458,14 +458,15 @@ train_ch3(net, train_iter, test_iter, cross_entropy, num_epochs, updater)
...
-## *dịch tiêu đề phía trên*
+## Dự đoán
 
 <!--
 Now that training is complete, our model is ready to classify some images.
 Given a series of images, we will compare their actual labels (first line of text output) and the model predictions (second line of text output).
 -->
 
-*dịch đoạn phía trên*
+Bây giờ khi việc huấn luyện hoàn thành, mô hình của chúng ta sẵn sàng để phân loại các ảnh.
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Giờ thì việc huấn luyện đã hoàn thành, mô hình của chúng ta đã sẵn sàng để phân loại các ảnh.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 2]
```diff
@@ -458,14 +458,15 @@ train_ch3(net, train_iter, test_iter, cross_entropy, num_epochs, updater)
...
+Cho trước một chuỗi các ảnh, chúng ta sẽ so sánh các nhãn thực của chúng (dòng đầu tiên của văn bản đầu ra) với những dự đoán của mô hình (dòng thứ hai của văn bản đầu ra).
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]: dùng từ `chuỗi` khiến mình nghĩ đến việc các ảnh được sắp xếp theo 1 thứ tự nào đó
```suggestion
Cho một loạt các ảnh, chúng ta sẽ so sánh các nhãn thực của chúng (dòng đầu tiên của văn bản đầu ra) với những dự đoán của mô hình (dòng thứ hai của văn bản đầu ra).
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 3]
```diff
@@ -510,7 +513,11 @@ What could be the problem with this implementation (hint: consider the domain of
 5. Assume that we want to use softmax regression to predict the next word based on some features. What are some problems that might arise from a large vocabulary?
 -->
 
-*dịch đoạn phía trên*
+1. Trong mục này, chúng ta đã thực hiện hàm softmax dựa vào các định nghĩa toán học của phép toán softmax. Điều này có thể gây ra những vấn đề gì (gợi ý: thử tính $\exp(50)$)?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
1. Trong mục này, chúng ta đã lập trình hàm softmax dựa vào định nghĩa toán học của phép toán softmax. Điều này có thể gây ra những vấn đề gì (gợi ý: thử tính $\exp(50)$)?
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 4]
```diff
@@ -510,7 +513,11 @@ What could be the problem with this implementation (hint: consider the domain of
...
+2. Hàm `cross_entropy` trong mục này được thực hiện dựa vào định nghĩa của hàm mất mát entropy chéo. Điều gì có thể là vấn đề của cách thực hiện này (gợi ý: xem xét miền của hàm log)?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
2. Hàm `cross_entropy` trong mục này được lập trình dựa vào định nghĩa của hàm mất mát entropy chéo. Vấn đề gì có thể xảy ra với cách lập trình như vậy (gợi ý: xem xét miền của hàm log)?
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 5]
```diff
@@ -510,7 +513,11 @@ What could be the problem with this implementation (hint: consider the domain of
...
+3. Hãy tìm các giải pháp để giải quyết hai vấn đề trên?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
3. Hãy tìm các giải pháp để giải quyết hai vấn đề trên.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 6]
```diff
@@ -510,7 +513,11 @@ What could be the problem with this implementation (hint: consider the domain of
...
+4. Trả về nhãn có khả năng nhất có phải luôn là một ý tưởng tốt. Ví dụ, bạn có dùng phương pháp này cho chẩn đoán bệnh hay không?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
4. Việc trả về nhãn có khả năng nhất có phải lúc nào cũng là ý tưởng tốt không? Ví dụ, bạn có dùng phương pháp này cho chẩn đoán bệnh hay không?
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 7]
```diff
@@ -484,21 +485,23 @@ predict_ch3(net, test_iter)
...
 
 <!--
 With softmax regression, we can train models for multi-category classification.
 The training loop is very similar to that in linear regression: retrieve and read data, define models and loss functions, then train models using optimization algorithms.
 As you will soon find out, most common deep learning models have similar training procedures.
 -->
 
-*dịch đoạn phía trên*
+Với hồi quy softmax, chúng ta có thể huấn luyện các mô hình cho bài toán phân loại đa lớp.
+Vòng lặp huấn luyện rất giống với vòng lặp trong hồi quy tuyến tính: truy xuất và đọc dữ liệu, đinh nghĩa các mô hình và các hàm mất mát, sau đó huấn luyện các mô hình này sử dụng các giải thuật tối ưu.
```
Comment (tanduong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Vòng lặp huấn luyện rất giống với vòng lặp huấn luyện của hồi quy tuyến tính: truy xuất và đọc dữ liệu, đinh nghĩa mô hình và hàm mất mát, và rồi huấn luyện các mô hình sử dụng các giải thuật tối ưu.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 8]
```diff
@@ -484,21 +485,23 @@ predict_ch3(net, test_iter)
...
+Như bạn sẽ sớm nhận thấy, hầu hết những mô hình học sâu phổ biến đều có thủ tục huấn luyện tương tự nhau.
```
Comment (tanduong) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
Rồi bạn sẽ thấy rằng hầu hết các mô hình học sâu phổ biến đều có thủ tục huấn luyện tương tự như vậy.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md [hunk 9]
```diff
@@ -510,7 +513,11 @@ What could be the problem with this implementation (hint: consider the domain of
 5. Assume that we want to use softmax regression to predict the next word based on some features. What are some problems that might arise from a large vocabulary?
 -->
 
-*dịch đoạn phía trên*
+1. Trong mục này, chúng ta đã thực hiện hàm softmax dựa vào các định nghĩa toán học của phép toán softmax. Điều này có thể gây ra những vấn đề gì (gợi ý: thử tính $\exp(50)$)?
+2. Hàm `cross_entropy` trong mục này được thực hiện dựa vào định nghĩa của hàm mất mát entropy chéo. Điều gì có thể là vấn đề của cách thực hiện này (gợi ý: xem xét miền của hàm log)?
+3. Hãy tìm các giải pháp để giải quyết hai vấn đề trên?
+4. Trả về nhãn có khả năng nhất có phải luôn là một ý tưởng tốt. Ví dụ, bạn có dùng phương pháp này cho chẩn đoán bệnh hay không?
+5. Giả sử rằng chúng ta muốn sử dụng hồi quy softmax để dự đoán từ tiếp theo dựa vào một số đặc trưng. Những vấn đề gì có thể xảy ra nếu dùng một tập từ vựng lớn?
```
Comment (minhduc0711) [inline chapter_linear-networks/softmax-regression-scratch_vn.md]:
```suggestion
5. Giả sử rằng chúng ta muốn sử dụng hồi quy softmax để dự đoán từ tiếp theo dựa vào một số đặc trưng. 
Những vấn đề gì có thể xảy ra nếu dùng một tập từ vựng lớn?
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 7 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: *sửa typo
```suggestion
Vòng lặp huấn luyện rất giống với vòng lặp huấn luyện của hồi quy tuyến tính: truy xuất và đọc dữ liệu, định nghĩa mô hình và hàm mất mát, và rồi huấn luyện mô hình sử dụng các giải thuật tối ưu.
```

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 9 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Cái này hông nên thêm dòng mới nhe.

Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: đây là quy ước của nhóm để việc review dễ hơn thôi, lúc render nó cũng không hiện dòng mới đâu

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 1 above)
Comment (azraelzhor) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: ok bạn

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 9 above)
Comment (duythanhvn) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: @minhduc0711 Chết mệ, này phải lưu ý với anh em cứ theo format rồi :)))

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 8 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Chỗ này `như vậy` đọc trôi chảy hơn `như nhau`. Mình chưa hiểu tại sao resolve mà chưa phản hồi.

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 2 above)
Comment (tanduong) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: Chỗ này cũng cần phản hồi từ bạn @azraelzhor.

Comment (azraelzhor) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: chỗ này ok nha

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 8 above)
Comment (azraelzhor) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: chỗ này ok nha bạn, sorry mình ma mới nên chưa quen :v

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 6 above)
Comment (azraelzhor) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: chỗ này cũng ok

File: chapter_linear-networks/softmax-regression-scratch_vn.md (hunk 8 above)
Comment (minhduc0711) [reply chapter_linear-networks/softmax-regression-scratch_vn.md]: nếu thấy ok bạn phải nhấn commit suggestion nó mới update nha @azraelzhor
